plugins/inputs/systemd_units/systemd_units_test.go

```diff
@@ -38,6 +39,52 @@ func TestDefaultPattern(t *testing.T) {
 	require.Equal(t, "*", plugin.Pattern)
 }
 
+func TestDefaultScope(t *testing.T) {
+	tests := []struct {
+		name     string
+		scope    string
+		expected map[string]string
+	}{
+		{
+			scope: "",
+			expected: map[string]string{
+				"scope": "system",
+				"user":  "",
+			},
+		},
+		{
+			scope: "system",
+			expected: map[string]string{
+				"scope": "system",
+				"user":  "",
+			},
+		},
+		{
+			scope: "user",
+			expected: map[string]string{
+				"scope": "user",
+				"user":  os.Getenv("USER"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := &SystemdUnits{
+				Scope: tt.scope,
+			}
+			require.NoError(t, plugin.Init())
+			if tt.expected["scope"] != plugin.scope {
+				t.Fatalf("Incorrect scope set: %q", plugin.scope)
+			}
+
+			if tt.expected["user"] != plugin.user {
+				t.Fatalf("Wrong username established: %q", plugin.user)
+			}
```

Comment:
Please use the power of the `require` package

```suggestion
			require.Equal(t, tt.expectedScope, plugin.scope)
			require.Equal(t, tt.expectedUser, plugin.user)
```

and avoid the map declaration for the expected values. Instead define them as separate members...